core/src/main/scala/sjsonnew/package.scala

```diff
@@ -23,6 +23,7 @@ package object sjsonnew {
 
   type LNil = LList.LNil0
   val LNil = LList.LNil0
+  type :*:[A1, A2 <: LList] = LCons[A1, A2]
```

Comment:
Weren't there a Scala compiler bug about splitting type alias and companion-looking object into different scope? https://github.com/scala/bug/issues/7139
A common workaround is to move the `object :*:` here. See https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/data/package.scala